*** usbmuxd-1.0.8/libusbmuxd/usbmuxd-proto.h	2012-04-08 00:50:09.000000000 +0200
--- usbmuxd_ironha1l/libusbmuxd/usbmuxd-proto.h	2012-07-31 10:52:14.000000000 +0200
***************
*** 31,37 ****
  #if defined(WIN32) || defined(__CYGWIN__)
  #define USBMUXD_SOCKET_PORT 27015
  #else
! #define USBMUXD_SOCKET_FILE "/var/run/usbmuxd"
  #endif
  
  #ifdef __cplusplus
--- 31,37 ----
  #if defined(WIN32) || defined(__CYGWIN__)
  #define USBMUXD_SOCKET_PORT 27015
  #else
! #define USBMUXD_SOCKET_FILE "/data/local/tmp/usbmuxd"
  #endif
  
  #ifdef __cplusplus
